WeatherForecast/Services/CalculationService.cs

```diff
@@ -121,5 +139,31 @@ public List<WeatherForecastDto> CalculateAverageMetrics(List<WeatherForecast> ho
                 throw new CalculationException("Oops something went wrong with calculation");
             }
         }
+
+        private string ModifyWeatherDescription(string description)
+        {
+            var modifiedDescription = string.Empty;
+
+            if (description.Contains("cloudy") || description.Contains("clouds"))
+            {
+                modifiedDescription = "cloudy";
+            }
+
+            else if (description.Contains("rain") || description.Contains("rainy"))
```

Comment:
Second check is redundant. `rainy` contains `rain`. Second condition will never be true when the first condition is false. 
```suggestion
            else if (description.Contains("rain"))
```